numpy/ma/tests/test_core.py

```diff
@@ -3803,6 +3803,23 @@ def test_meananom_object(self):
         assert_equal(a.mean(), 2)
         assert_equal(a.anom(), [-1, 0, 1])
 
+    def test_anom_shape(self):
+        a = masked_array([1, 2, 3])
+        assert_equal(a.anom().shape, a.shape)
+        a.mask = True
+        assert_equal(a.anom().shape, a.shape)
+        assert_(np.ma.is_masked(a.anom()))
+
+    def test_anom(self):
+        a = masked_array(np.arange(1, 7).reshape(2, 3))
+        assert_equal(a.anom(), [[-2.5, -1.5, -0.5], [0.5, 1.5, 2.5]])
+        assert_equal(a.anom(axis=0), [[-1.5, -1.5, -1.5], [1.5, 1.5, 1.5]])
+        assert_equal(a.anom(axis=1), [[-1., 0., 1.], [-1., 0., 1.]])
+        a.mask = [[0, 0, 1], [0, 1, 0]]
+        assert_equal(format(a.anom()), '[[-2.25 -1.25 --]\n [0.75 -- 2.75]]')
```

Comment:
The only tests that use string comparison are `test_format` and `test_str_repr`. For those it makes sense, but here we are wanting to test numerical values rather than string representation. So let's change this to a more robust way.

The style in this file isn't great unfortunately, the regular way is testing the values - after filling the masked ones with a known value like `0` or `-9999` - and testing the mask separately. A bit weird, because the testing functions are specific to masked arrays (they come from `numpy/ma/testutils.py`).

Also, since these are floating-point numbers, we should use `assert_almost_equal` rather than `assert_equal` - as a rule, floating point numbers shouldn't be compared for exact equality because different hardware or compilers may give very small changes in results.